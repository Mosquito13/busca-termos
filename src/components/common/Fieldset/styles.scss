.fieldset {
  border: none;
  border-top: 1px solid currentColor;
  margin-top: 2rem;
  padding-top: 2rem;

  &--no-margin-top {
    margin-top: 0;
  }

  &__title {
    font-size: 1.6rem;
    text-align: left;
    padding: 0;
    padding-right: 2rem;
  }
}
