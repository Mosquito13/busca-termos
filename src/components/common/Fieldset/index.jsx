import classNames from 'classnames';
import PropTypes from 'prop-types';

import './styles.scss';

const Fieldset = ({ noMarginTop, title, children }) => {
  const classes = classNames(
    'fieldset',
    noMarginTop && 'fieldset--no-margin-top'
  );

  return (
    <fieldset className={classes}>
      <legend className="fieldset__title">{title}</legend>
      {children}
    </fieldset>
  );
};

Fieldset.propTypes = {
  /**
   * Titulo do fieldset
   */
  title: PropTypes.string,
  /**
   * Conteudo
   */
  children: PropTypes.node,
  /**
   * Indica se nao vai ter margem superior
   */
  noMarginTop: PropTypes.bool
};

export default Fieldset;
