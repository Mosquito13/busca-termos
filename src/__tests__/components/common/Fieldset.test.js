import renderer from 'react-test-renderer';

import Fieldset from '../../../components/common/Fieldset';

describe(`${Fieldset.name}`, () => {
  describe('snapshots', () => {
    it('should render normally', () => {
      const tree = renderer
        .create(
          <Fieldset>
            <div />
          </Fieldset>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render without margin top', () => {
      const tree = renderer
        .create(
          <Fieldset noMarginTop>
            <div />
          </Fieldset>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
