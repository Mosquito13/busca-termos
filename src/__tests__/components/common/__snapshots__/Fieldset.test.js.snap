// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset snapshots should render normally 1`] = `
<fieldset
  className="fieldset"
>
  <legend
    className="fieldset__title"
  />
  <div />
</fieldset>
`;

exports[`Fieldset snapshots should render without margin top 1`] = `
<fieldset
  className="fieldset fieldset--no-margin-top"
>
  <legend
    className="fieldset__title"
  />
  <div />
</fieldset>
`;
